lib/puppet/type/notify.rb

```diff
@@ -6,7 +6,7 @@ module Puppet
   Type.newtype(:notify) do
     @doc = "Sends an arbitrary message to the agent run-time log."
 
-    newproperty(:message) do
+    newproperty(:message, :idempotent => false) do
```

Comment:
You raise a good point @adrienthebo - without a mechanism for extending the whitelist, type and provider builders can't opt into the "non-idempotent" property behavior. Hmm...